api/src/test/java/uk/gov/dhsc/htbhf/claimant/message/processor/DetermineEntitlementMessageProcessorTest.java

```diff
@@ -146,16 +224,162 @@ void shouldUpdateClaimWhenClaimantIsNotEligible(EligibilityStatus eligibilitySta
-                context.getPreviousPaymentCycle());
+        runCommonVerifications(context, decision, message, messageStatus, ClaimStatus.PENDING_EXPIRY);
+        verify(determineEntitlementNotificationHandler).sendClaimNoLongerEligibleEmail(context.getClaim());
+        verifyZeroInteractions(pregnancyEntitlementCalculator, childDateOfBirthCalculator);
+    }
 
-        verify(paymentCycleService).updatePaymentCycle(context.getCurrentPaymentCycle(), decision);
-        verifyClaimSavedAtStatus(ClaimStatus.PENDING_EXPIRY);
+    //HTBHF-1757 No children in current cycle, are pregnant but ineligible from DWP
+    @ParameterizedTest(name = "Children DOB in previous cycle={0}")
+    @MethodSource("provideArgumentsForChildrenInPreviousCycle")
+    void shouldUpdateClaimToPendingExpiryAndSendNoLongerEligibleEmailWhenClaimantIsNotEligibleWithNoChildrenInCurrentCycle(
+            List<LocalDate> previousCycleChildrenDobs) {
+        //Given
+        List<LocalDate> currentCycleChildrenDobs = NO_CHILDREN;
+        LocalDate currentCycleExpectedDeliveryDate = EXPECTED_DELIVERY_DATE_IN_TWO_MONTHS;
+        LocalDate previousCycleExpectedDeliveryDate = EXPECTED_DELIVERY_DATE_IN_TWO_MONTHS;
+        DetermineEntitlementMessageContext context = buildMessageContext(
+                previousCycleExpectedDeliveryDate,
+                currentCycleExpectedDeliveryDate,
+                previousCycleChildrenDobs,
+                currentCycleChildrenDobs);
+        given(messageContextLoader.loadDetermineEntitlementContext(any())).willReturn(context);
+        given(pregnancyEntitlementCalculator.isEntitledToVoucher(any(), any())).willReturn(true);
+
+        //Eligibility - any status that isn't ELIGIBLE has the same effect here.
+        EligibilityAndEntitlementDecision decision = aDecisionWithStatusAndChildren(INELIGIBLE, NOT_CONFIRMED, currentCycleChildrenDobs);
+        given(eligibilityAndEntitlementService.evaluateExistingClaimant(any(), any(), any())).willReturn(decision);
+
+        //Current payment cycle voucher entitlement mocking
+        Message message = aValidMessageWithType(DETERMINE_ENTITLEMENT);
+
+        //When
+        MessageStatus messageStatus = processor.processMessage(message);
+
+        //Then
+        runCommonVerifications(context, decision, message, messageStatus, ClaimStatus.PENDING_EXPIRY);
+        verify(determineEntitlementNotificationHandler).sendClaimNoLongerEligibleEmail(context.getClaim());
+        verify(pregnancyEntitlementCalculator).isEntitledToVoucher(EXPECTED_DELIVERY_DATE_IN_TWO_MONTHS, context.getCurrentPaymentCycle().getCycleStartDate());
+        verifyZeroInteractions(childDateOfBirthCalculator);
+    }
+
+    //HTBHF-1757 Claimant had children in previous cycle who would be under 4 in this cycle but are not returned from DWP
+    @Test
+    void shouldUpdateClaimToPendingExpiryAndSendNoLongerEligibleEmailWhenClaimantIsNotEligibleWithNoChildrenInCurrentCycle() {
```

Comment:
Could we have a display name for this test and the one above. Otherwise it's quiet difficult to see where they differ, especially as they have the same name.

Reply:
Sorted now.